pkg/daemon/certificate_writer.go

```diff
@@ -102,37 +118,49 @@ func (dn *Daemon) syncControllerConfigHandler(key string) error {
 	// Also annotate the latest config we've seen, so as to not write unnecessarily
 	currentNodeControllerConfigResource := dn.node.Annotations[constants.ControllerConfigResourceVersionKey]
 
-	if currentNodeControllerConfigResource != controllerConfig.ObjectMeta.ResourceVersion {
+	if currentNodeControllerConfigResource != controllerConfig.ObjectMeta.ResourceVersion || controllerConfig.Annotations["machineconfiguration.openshift.io/sync-service-ca"] == "true" {
 		pathToData := make(map[string][]byte)
 
 		kubeAPIServerServingCABytes := controllerConfig.Spec.KubeAPIServerServingCAData
 		cloudCA := controllerConfig.Spec.CloudProviderCAData
 		pathToData[caBundleFilePath] = kubeAPIServerServingCABytes
 		pathToData[cloudCABundleFilePath] = cloudCA
+		var data []byte
+		var err error
 
-		for bundle, data := range pathToData {
-			if !strings.HasSuffix(string(data), "\n") {
-				bString := string(data) + "\n"
-				data = []byte(bString)
-			}
-			if Finfo, err := os.Stat(bundle); err == nil {
-				var mode os.FileMode
-				Dinfo, err := os.Stat(filepath.Dir(bundle))
+		cm, cmErr := dn.kubeClient.CoreV1().ConfigMaps("openshift-machine-config-operator").Get(context.TODO(), "kubeconfig-data", v1.GetOptions{})
+		if cmErr != nil {
+			klog.Errorf("kubeconfig-data does not exist yet. %v", cmErr)
```

Comment:
question: If the error returned indicates that a different API error occurred (e.g., API server unavailable, improper permissions, etc.), should this still assume that `kubeconfig-data` does not exist yet?

Reply:
good point, I guess I can check for that